cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceStatusTest.java

```diff
@@ -0,0 +1,202 @@
+
+    }
+
+    @Test
+    public void testCreateLoadMap() {
+
+        JsonObject proposal = buildOptimizationProposal();
+
+        JsonArray loadBeforeArray = proposal.getJsonObject(CruiseControlRebalanceKeys.LOAD_BEFORE_OPTIMIZATION.getKey())
+                .getJsonArray(CruiseControlRebalanceKeys.BROKERS.getKey());
+        JsonArray loadAfterArray = proposal.getJsonObject(CruiseControlRebalanceKeys.LOAD_AFTER_OPTIMIZATION.getKey())
+                .getJsonArray(CruiseControlRebalanceKeys.BROKERS.getKey());
+
+        JsonObject output = KafkaRebalanceAssemblyOperator.parseLoadStats(
+                loadBeforeArray, loadAfterArray);
+
+        try {
+            ObjectMapper mapper = new ObjectMapper();
+            Map<String, LinkedHashMap<String, String>> outputMap = mapper.readValue(output.encodePrettily(), Map.class);
+            assertThat(outputMap, hasKey("1"));
+
+            LinkedHashMap<String, LinkedHashMap> m = (LinkedHashMap) outputMap.get("1");
+
+            assertThat(outputMap.get("1"), hasKey(CruiseControlLoadParameters.CPU_PERCENTAGE.getKafkaRebalanceStatusKey()));
+
+            ArrayList<Integer> replicas = new ArrayList<>();
+            replicas.add((Integer) m.get("replicas").get("before"));
+            replicas.add((Integer) m.get("replicas").get("after"));
+            replicas.add((Integer) m.get("replicas").get("diff"));
+
+            assertThat(replicas.get(0), is(10));
+            assertThat(replicas.get(1), is(5));
+            assertThat(replicas.get(2), is(-5));
+
+            assertThat(outputMap.get("1"), hasKey(CruiseControlLoadParameters.REPLICAS.getKafkaRebalanceStatusKey()));
+
+            ArrayList<Double> cpu = new ArrayList<>();
+            cpu.add((Double) m.get("cpuPercentage").get("before"));
+            cpu.add((Double) m.get("cpuPercentage").get("after"));
+            cpu.add((Double) m.get("cpuPercentage").get("diff"));
+
+            assertThat(cpu.get(0), is(10.));
+            assertThat(cpu.get(1), is(20.0));
+            assertThat(cpu.get(2), is(10.0));
+
+        } catch (JsonProcessingException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    public void testProcessProposal() {
+
+        JsonObject proposal = buildOptimizationProposal();
+
+        KafkaRebalance kr =
+                createKafkaRebalance(CLUSTER_NAMESPACE, CLUSTER_NAME, RESOURCE_NAME, new KafkaRebalanceSpecBuilder().build());
+
+        KafkaRebalanceAssemblyOperator.MapandStatus<ConfigMap, Map<String, Object>> output = KafkaRebalanceAssemblyOperator.processOptimizationProposal(kr, proposal);
+
+        assertTrue(output.getStatus().containsKey(CruiseControlRebalanceKeys.SUMMARY.getKey()));
+
+        Map<String, String> brokerMap = output.getLoadMap().getData();
+
+        try {
+
+            ObjectMapper mapper = new ObjectMapper();
+
+            Map<String, LinkedHashMap<String, String>> brokerLoadMap = mapper.readValue(brokerMap.get("brokerLoad"), LinkedHashMap.class);
```

Comment:
Similar to above you should test that the ConfigMap you were given has the data in you expected. You can test its fields directly, you don't need to force into a map.